src/common/argscheck.js

```diff
@@ -36,6 +36,35 @@ function extractParamName (callee, argIndex) {
     return (/\(\s*([^)]*?)\s*\)/).exec(callee)[1].split(/\s*,\s*/)[argIndex];
 }
 
+/**
+ * Throws TypeError if the provided arguments do not have the expected types.
+ *
+ * `spec` is a string where each character stands for the required type of the
+ * argument at the same position. In other words: the character at `spec[i]`
+ * specifies the required type for `args[i]`. The characters in `spec` are the
+ * first letter of the required type's name. The supported types are:
+ *
+ *     Array, Date, Number, String, Function, Object
+ *
+ * Lowercase characters specify arguments that must not be `null` or `undefined`
+ * while uppercase characters allow those values to be passed.
+ *
+ * Finally, `*` can be used to allow any type at the corresponding position.
+ *
+ * @example
+ * function foo (arr, opts) {
+ *     // require `arr` to be an Array and `opts` an Object, null or undefined
+ *     checkArgs('aO', 'my.package.foo', arguments);
```

Comment:
```suggestion
 *     checkArgs('aO', 'my.package.foo', opts);
```
I guess? 
If not, and `arguments` would be created inside `foo` somewhere first, then maybe add `...` do indicate left out code.

Reply:
No, the example is correct. [`arguments` is special](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments)

Reply:
TIL